src/components/Root.tsx

```diff
@@ -1,30 +1,75 @@
+import createCache from '@emotion/cache';
+import { CacheProvider } from '@emotion/react';
+import React from 'react';
 import { I18nextProvider } from 'react-i18next';
 import { ToastContainer } from 'react-toastify';
 import 'react-toastify/dist/ReactToastify.css';
+import { prefixer } from 'stylis';
+import rtlPlugin from 'stylis-plugin-rtl';
 
 import { theme } from '@graasp/ui';
 
+import { MenuItem, Select } from '@mui/material';
 import { ThemeProvider } from '@mui/material/styles';
 
 import { SHOW_NOTIFICATIONS } from '../config/constants';
 import i18nConfig from '../config/i18n';
+import i18n from '../config/i18n';
 import {
   QueryClientProvider,
   ReactQueryDevtools,
   queryClient,
 } from '../config/queryClient';
 import App from './App';
 
-const Root = () => (
-  <QueryClientProvider client={queryClient}>
-    <I18nextProvider i18n={i18nConfig}>
-      <ThemeProvider theme={theme}>
-        {SHOW_NOTIFICATIONS && <ToastContainer />}
-        <App />
-      </ThemeProvider>
-    </I18nextProvider>
-    {process.env.NODE_ENV === 'development' && <ReactQueryDevtools />}
-  </QueryClientProvider>
-);
+const cacheRtl = createCache({
+  key: 'muirtl',
+  stylisPlugins: [prefixer, rtlPlugin],
+});
+
+const cacheLtr = createCache({
+  key: 'muiltr',
+  stylisPlugins: [prefixer],
+});
+
+const ltrTheme = { ...theme, direction: 'ltr' };
+const rtlTheme = { ...theme, direction: 'rtl' };
+
+const Root = () => {
+  const [curr_lang, setCurr_lang] = React.useState('en');
+
+  const [isRtl, setIsRtl] = React.useState(true);
+
+  const handleLangSelect = (event) => {
+    setIsRtl(curr_lang === 'ar');
+    isRtl ? i18n.changeLanguage('ar') : i18n.changeLanguage('en');
+    isRtl ? (document.dir = 'rtl') : (document.dir = 'ltr');
+    setCurr_lang(event.target.value);
+  };
+
+  return (
+    <QueryClientProvider client={queryClient}>
+      <I18nextProvider i18n={i18nConfig}>
+        <CacheProvider value={isRtl ? cacheLtr : cacheRtl}>
+          <ThemeProvider theme={isRtl ? rtlTheme : ltrTheme}>
+            {/*/!*@ts-ignore*!/*/}
+            {SHOW_NOTIFICATIONS && <ToastContainer />}
+            <Select
+              label="Change Language"
+              value={curr_lang}
+              onChange={handleLangSelect}
+            >
+              <MenuItem value={'en'}>{'en'}</MenuItem>
+              <MenuItem value={'ar'}>{'ar'}</MenuItem>
+            </Select>
```

Comment:
As mentioned, use `@graasp/translations`'s `langs` to get all the available languages.